src/dtos/users.dto.ts

```diff
@@ -1,9 +1,20 @@
-import { IsEmail, IsString } from 'class-validator';
+export type BaseUserDto = {
+  readonly id: number;
+  readonly email: string;
+  readonly password: string;
+  readonly firstName: string;
+  readonly lastName: string;
+  readonly language: string;
+  readonly profilePic?: string;
+  readonly dateOfBirth?: Date;
+}
 
-export class CreateUserDto {
-  @IsEmail()
-  public email: string;
+export type LoginUserRequestDto = Pick<BaseUserDto, "email" | "password">
 
-  @IsString()
-  public password: string;
-}
+export type RegisterUserRequestDto = Omit<BaseUserDto, "id">
+
+export type AuthenticateUserRequestDto = Pick<BaseUserDto, "id">
```

Comment:
Do we need Authentication DTO at all? Are we going to store user ID in JWT payload?